core/templates/services/math-interactions.service.ts

```diff
@@ -183,6 +183,102 @@ export class MathInteractionsService {
     }
     return modifiedExpressionList.join('');
   }
+
+  getTerms(expressionString: string, splitByAddition = true): Array<string> {
+    let listOfTerms: Array<string> = [];
+    let currentTerm: string = '';
+    let bracketBalance: number = 0;
+    let shouldModifyNextTerm: boolean = false;
+    let modifyTerm = function(termString: string): string {
+      // If the shouldModifyNextTerm flag is set to true, we add the '-' sign,
+      // raise the term to a power of -1. This ensures that when the final
+      // list is joined by the '+'/'*' sign, it matches with the original
+      // expression. For eg.
+      // '3/10' would be split as [3, 10^(-1)] and
+      // '3-10' would be split as [3, -(10)].
+      if (splitByAddition) {
+        return '-(' + termString + ')';
+      } else {
+        return '(' + termString + ')^(-1)';
+      }
+    };
+    const primaryDelimiter = splitByAddition ? '+' : '*';
+    const secondaryDelimiter = splitByAddition ? '-' : '/';
+
+    expressionString = expressionString.replace(/\s/g, '');
+
+    // Temporarily replacing all unary negation signs with '~' so as to avoid
+    // splitting terms by them. We only need to split terms by binary
+    // subtraction signs and not unary negation signs. A '-' sign is considered
+    // to be binary subtraction iff it is preceded by an alphanumeric or a
+    // closing bracket, otherwise it is considered as unary negation operation.
+    // NOTE: The replace function is called twice to deal with cases where there
+    // might be overlapping matches.
+    // For eg. 4----5 would be converted to 4-~-~5 after the first call. So we
+    // need a second call to convert it to the desired result, which is 4-~~~5.
```

Comment:
Just to check, tests are there for the various ways in which this can occur, right?

Reply:
Yes.